api/src/services/user-service.ts

```diff
@@ -119,6 +133,34 @@ export class UserService extends DBService {
     return this.getUserById(userObject.id);
   }
 
+  /**
+   * Gets a system user, adding them if they do not already exist.
+   *
+   * @param {string} userGuid
+   * @param {string} userIdentifier
+   * @param {string} identitySource
+   * @return {*}  {Promise<Models.user.UserObject>}
+   * @memberof UserService
+   */
+  async getOrCreateSystemUser(
+    userGuid: string,
+    userIdentifier: string,
+    identitySource: string
+  ): Promise<Models.user.UserObject> {
+    defaultLog.debug({ label: 'getUserByGuid', userGuid, userIdentifier, identitySource });
+
+    // Check if the user exists
+    let userObject = await this.getUserByGuid(userGuid);
+
+    if (!userObject) {
+      // Found no existing user, add them
+      userObject = await this.addSystemUser(userGuid, userIdentifier, identitySource);
+    }
+
+    // get the newly created user
+    return this.getUserById(userObject.id);
```

Comment:
Do we need this call, or can we just return the `userObject` we already have from either of the above two calls?